### ISSUE TITLE
graylog-collector-sidecar.exe unable to fetch config 404

### ISSUE THREAD

**Issue by fsdaniel** [Graylog2/collector-sidecar#156]
### Problem description

Windows collector-sidecar returns 404 upon trying to get configuration from graylog api.

### Steps to reproduce the problem

1. Have a installed and configured graylog server
2. create sidecar config on server (TAG: windows)
3. Install windows server 2012R2
OS Name:                   Microsoft Windows Server 2012 R2 Datacenter
OS Version:                6.3.9600 N/A Build 9600
4. edit collector_sidecar.yml to work properly

### Environment

* Sidecar Version: 0.1.0-1
* Graylog Version: 2.2.2+691b4b7
* Operating System: Ubuntu 14.04.5 LTS
* Elasticsearch Version: 2.3.3
* MongoDB Version: v2.6.12

### Log
```
time="2017-03-28T10:58:44+02:00" level=debug msg="Add collector backend: winlogbeat" 
time="2017-03-28T10:58:44+02:00" level=info msg="Starting signal distributor" 
time="2017-03-28T10:58:44+02:00" level=debug msg="[signal-processor] (seq=1) handling cmd: restart" 
time="2017-03-28T10:58:44+02:00" level=info msg="[winlogbeat] Starting (exec driver)" 
time="2017-03-28T10:58:44+02:00" level=debug msg="Creating rotated log writer for: C:\\Program Files\\graylog\\collector-sidecar\\logs\\winlogbeat_stderr.log.%Y%m%d%H%M" 
time="2017-03-28T10:58:44+02:00" level=debug msg="Creating rotated log writer for: C:\\Program Files\\graylog\\collector-sidecar\\logs\\winlogbeat_stderr.log.%Y%m%d%H%M" 
time="2017-03-28T10:58:44+02:00" level=debug msg="[signal-processor] (seq=1) cmd done: restart" 
time="2017-03-28T10:58:45+02:00" level=error msg="[winlogbeat] Backend finished unexpectedly, trying to restart 1/3." 
time="2017-03-28T10:58:45+02:00" level=debug msg="[signal-processor] (seq=2) handling cmd: restart" 
time="2017-03-28T10:58:45+02:00" level=info msg="[winlogbeat] Stopping" 
time="2017-03-28T10:58:47+02:00" level=debug msg="[winlogbeat] SIGHUP ignored, killing process" 
time="2017-03-28T10:58:47+02:00" level=debug msg="[winlogbeat] Failed to kill process invalid argument" 
time="2017-03-28T10:58:47+02:00" level=info msg="[winlogbeat] Starting (exec driver)" 
time="2017-03-28T10:58:47+02:00" level=debug msg="Creating rotated log writer for: C:\\Program Files\\graylog\\collector-sidecar\\logs\\winlogbeat_stderr.log.%Y%m%d%H%M" 
time="2017-03-28T10:58:47+02:00" level=debug msg="Creating rotated log writer for: C:\\Program Files\\graylog\\collector-sidecar\\logs\\winlogbeat_stderr.log.%Y%m%d%H%M" 
time="2017-03-28T10:58:47+02:00" level=debug msg="[signal-processor] (seq=2) cmd done: restart" 
time="2017-03-28T10:58:48+02:00" level=error msg="[winlogbeat] Backend finished unexpectedly, trying to restart 2/3." 
time="2017-03-28T10:58:48+02:00" level=debug msg="[signal-processor] (seq=3) handling cmd: restart" 
time="2017-03-28T10:58:48+02:00" level=info msg="[winlogbeat] Stopping" 
time="2017-03-28T10:58:50+02:00" level=debug msg="[winlogbeat] SIGHUP ignored, killing process" 
time="2017-03-28T10:58:50+02:00" level=debug msg="[winlogbeat] Failed to kill process invalid argument" 
time="2017-03-28T10:58:50+02:00" level=info msg="[winlogbeat] Starting (exec driver)" 
time="2017-03-28T10:58:50+02:00" level=debug msg="Creating rotated log writer for: C:\\Program Files\\graylog\\collector-sidecar\\logs\\winlogbeat_stderr.log.%Y%m%d%H%M" 
time="2017-03-28T10:58:50+02:00" level=debug msg="Creating rotated log writer for: C:\\Program Files\\graylog\\collector-sidecar\\logs\\winlogbeat_stderr.log.%Y%m%d%H%M" 
time="2017-03-28T10:58:50+02:00" level=debug msg="[signal-processor] (seq=3) cmd done: restart" 
time="2017-03-28T10:58:51+02:00" level=error msg="[winlogbeat] Backend finished unexpectedly, trying to restart 3/3." 
time="2017-03-28T10:58:51+02:00" level=debug msg="[signal-processor] (seq=4) handling cmd: restart" 
time="2017-03-28T10:58:51+02:00" level=info msg="[winlogbeat] Stopping" 
time="2017-03-28T10:58:53+02:00" level=debug msg="[winlogbeat] SIGHUP ignored, killing process" 
time="2017-03-28T10:58:53+02:00" level=debug msg="[winlogbeat] Failed to kill process invalid argument" 
time="2017-03-28T10:58:53+02:00" level=info msg="[winlogbeat] Starting (exec driver)" 
time="2017-03-28T10:58:53+02:00" level=debug msg="Creating rotated log writer for: C:\\Program Files\\graylog\\collector-sidecar\\logs\\winlogbeat_stderr.log.%Y%m%d%H%M" 
time="2017-03-28T10:58:53+02:00" level=debug msg="Creating rotated log writer for: C:\\Program Files\\graylog\\collector-sidecar\\logs\\winlogbeat_stderr.log.%Y%m%d%H%M" 
time="2017-03-28T10:58:53+02:00" level=debug msg="[signal-processor] (seq=4) cmd done: restart" 
time="2017-03-28T10:58:54+02:00" level=error msg="[RequestConfiguration] Bad response status from Graylog server: 404 Not Found" 
time="2017-03-28T10:58:54+02:00" level=error msg="[UpdateRegistration] Bad response from Graylog server: 404 Not Found" 
time="2017-03-28T10:58:54+02:00" level=error msg="Can't fetch configuration from Graylog API: GET http://xxxxxxxxxx:9000/api/plugins/org.graylog.plugins.collector/154537d3-868c-4ccc-a8c7-9ca3ec733d1d?tags=%5B%22windows%22%5D: 404 HTTP 404 Not Found" 
time="2017-03-28T10:58:54+02:00" level=error msg="[winlogbeat] Unable to start collector after 3 tries, giving up!" 
time="2017-03-28T10:59:04+02:00" level=error msg="[RequestConfiguration] Bad response status from Graylog server: 404 Not Found" 
time="2017-03-28T10:59:04+02:00" level=error msg="Can't fetch configuration from Graylog API: GET http://xxxxxxxxxx:9000/api/plugins/org.graylog.plugins.collector/154537d3-868c-4ccc-a8c7-9ca3ec733d1d?tags=%5B%22windows%22%5D: 404 HTTP 404 Not Found" 
time="2017-03-28T10:59:04+02:00" level=error msg="[UpdateRegistration] Bad response from Graylog server: 404 Not Found" 
time="2017-03-28T10:59:14+02:00" level=error msg="[RequestConfiguration] Bad response status from Graylog server: 404 Not Found" 
time="2017-03-28T10:59:14+02:00" level=error msg="Can't fetch configuration from Graylog API: GET http://xxxxxxxxxx:9000/api/plugins/org.graylog.plugins.collector/154537d3-868c-4ccc-a8c7-9ca3ec733d1d?tags=%5B%22windows%22%5D: 404 HTTP 404 Not Found" 
time="2017-03-28T10:59:14+02:00" level=error msg="[UpdateRegistration] Bad response from Graylog server: 404 Not Found" 
time="2017-03-28T10:59:14+02:00" level=info msg="Stopping signal distributor" 
```

### Manual fetch attempt
```
PS C:\temp> Invoke-WebRequest -uri http://xxxxxxxxxx:9000/api/plugins/org.graylog.plugins.collector/a0c
b6eb7-b757-4af6-97a2-ef2e85a1dd03?tags=%5B%22windows%22%5D


StatusCode        : 200
StatusDescription : OK
Content           : <!DOCTYPE html>
                    <html>
                      <head>
                        <meta http-equiv="X-UA-Compatible" content="IE=edge">
                        <meta name="robots" content="noindex, nofollow">
                        <meta charset="UTF-8">
                        <title>Graylog Web Interfa...
RawContent        : HTTP/1.1 200 OK
                    X-Graylog-Node-ID: 3e5894a7-11d7-4c11-b1c1-dcce30f0bdb8
                    X-UA-Compatible: IE=edge
                    Content-Length: 1290
                    Content-Type: text/html
                    Date: Tue, 28 Mar 2017 08:47:00 GMT

                    <!DOCTYPE html...
Forms             : {}
Headers           : {[X-Graylog-Node-ID, 3e5894a7-11d7-4c11-b1c1-dcce30f0bdb8], [X-UA-Compatible, IE=edge], [Content-Le
                    ngth, 1290], [Content-Type, text/html]...}
Images            : {}
InputFields       : {}
Links             : {}
ParsedHtml        : mshtml.HTMLDocumentClass
RawContentLength  : 1290
```

### collector_sidecar.yml
```
server_url: http://xxxxxxxxxxxxxxxx:9000/api/ 
update_interval: 10
tls_skip_verify: true
send_status: true
list_log_files:
node_id: graylog-collector-sidecar
collector_id: file:C:\Program Files\graylog\collector-sidecar\collector-id
cache_path: C:\Program Files\graylog\collector-sidecar\cache
log_path: C:\Program Files\graylog\collector-sidecar\logs
log_rotation_time: 86400
log_max_age: 604800
tags: [windows]
backends:
    - name: winlogbeat
      enabled: true
      binary_path: C:\Program Files\graylog\collector-sidecar\winlogbeat.exe
      configuration_path: C:\Program Files\graylog\collector-sidecar\generated\winlogbeat.yml
```

**Comment by joschi** [comment]
@fsdaniel Are you sure that your `server_url` is correct? It doesn't look like it from the HTTP response you've mentioned.

**Comment by fsdaniel** [comment]
Am i looking in the wrong place ?

I have omitted the real URL and replaced with xxxxxxxxxx.

From the sidecar log:
```
GET http://xxxxxxxxxx:9000/api/plugins/org.graylog.plugins.collector/154537d3-868c-4ccc-a8c7-9ca3ec733d1d?tags=%5B%22windows%22%5D: 404 HTTP 404 Not Found"
```

Our manual attempt calls this url also and gets a HTTP 200.

**Comment by joschi** [comment]
@fsdaniel What's the output of the following curl commands (replace credentials with your own and use the correct URL):
```
# curl -u $GL_USER:$GL_PASSWORD -H 'Accept: application/json' http://xxxxxxxxxx:9000/api/system?pretty=true
# curl -u $GL_USER:$GL_PASSWORD -H 'Accept: application/json' http://xxxxxxxxxx:9000/api/plugins/org.graylog.plugins.collector/154537d3-868c-4ccc-a8c7-9ca3ec733d1d?tags=%5B%22windows%22%5D
# curl -u $GL_USER:$GL_PASSWORD -H 'Accept: application/json' http://xxxxxxxxxx:9000/api/plugins/org.graylog.plugins.collector/154537d3-868c-4ccc-a8c7-9ca3ec733d1d?tags=windows
```

You might also want to review the value for the `tags` setting in your `collector_sidecar.yml` file.

**Comment by fsdaniel** [comment]
@joschi 

In all cases this resolves to:
```
{"type":"ApiError","message":"HTTP 404 Not Found"}
```

I went and looked in the server config for any other rest related ports found something to be at 12900 and queried this also with the same result.

I dug into the documentation as when i queried the same URL (api/system/) on my other graylog system i was provided info that i expected.

Reading http://docs.graylog.org/en/2.2/pages/configuration/web_interface.html

I adjusted the configuration in server.conf for this server from api rest call

```
rest_listen_uri = http://xxxxxxxxxx:12900/
```
to
```
rest_listen_uri = http://xxxxxxxxxx:9000/api/
```

This has resolved the issue of the 404's. 

Seems this is a case of 'RTFM' during upgrades on my part, there for my apology.

Thank you for providing the step to start the chain to success.
